Il est important que les sociétés de cartes de crédit soient capables de reconnaître les transactions frauduleuses afin que les clients ne soient pas facturés pour des articles qu'ils n'ont pas achetés.

 Projet 4 : Détection de fraude

## Contenu

L'ensemble de données contient les transactions par carte bancaire effectuées en septembre 2013 par des titulaires européens.
Il présente les transactions réalisées sur deux jours et recense 492 fraudes sur 284 807 transactions. L'ensemble de données est fortement déséquilibré : les transactions frauduleuses ne représentent que 0,172 % du total.

Ce jeu de données contient uniquement des variables d'entrée numériques, issues d'une transformation ACP. Malheureusement, pour des raisons de confidentialité, nous ne pouvons fournir ni les variables d'origine ni d'informations complémentaires sur les données. Les variables V1, V2, …, V28 correspondent aux composantes principales obtenues par ACP. Seules les variables « Temps » et « Montant » n'ont pas été transformées par ACP. La variable « Temps » représente le temps écoulé en secondes entre chaque transaction et la première transaction du jeu de données. La variable « Montant » correspond au montant de chaque transaction ; elle peut être utilisée pour un apprentissage sensible aux coûts et dépendant des exemples. La variable « Classe » est la variable réponse : elle prend la valeur 1 en cas de fraude et 0 sinon.

Compte tenu du déséquilibre des classes, nous recommandons de mesurer la précision à l'aide de l'aire sous la courbe précision-rappel (AUPRC). La précision de la matrice de confusion n'est pas pertinente pour une classification déséquilibrée.

 1.  1. Importation des bibliothèques

Pour réaliser l’analyse exploratoire des données (EDA), nous importons les bibliothèques Python nécessaires à la manipulation, à l’analyse et à la visualisation des données.

NumPy (`numpy`) est utilisé pour les opérations numériques et la manipulation de données sous forme de tableaux.
Pandas(`pandas`) permet de charger, manipuler et analyser le jeu de données sous forme de DataFrame.
Matplotlib (`matplotlib.pyplot`) permet de réaliser des visualisations graphiques afin de mieux comprendre les caractéristiques et la distribution des données.

Ces bibliothèques constituent la base des différentes étapes d’exploration qui seront réalisées par la suite.



In [2]:
# importation des bibliotheque
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

 2. Chargement des données

Après avoir importé les bibliothèques nécessaires, nous chargeons le jeu de données dans un DataFrame Pandas.

Le jeu de données utilisé dans ce projet contient des transactions bancaires et la variable cible `Class`, qui permet d'identifier les transactions normales (`0`) et frauduleuses (`1`).

Le chargement des données constitue une étape préalable à l'analyse exploratoire. Il nous permettra ensuite d'étudier la structure du jeu de données, ses dimensions, ses variables, ses types de données, ses valeurs manquantes, ses doublons et ses principales caractéristiques statistiques.

In [8]:
df = pd.read_csv("../data/creditcard.csv")

 3. Aperçu des premières lignes

Après avoir chargé le jeu de données, nous affichons les premières observations afin d’obtenir un premier aperçu de son contenu.

Cette étape permet de vérifier que les données ont été correctement chargées et d’observer les premières valeurs des différentes variables présentes dans le jeu de données. Elle constitue une première vérification avant d’analyser plus en détail sa structure et ses caractéristiques.


In [9]:
df.head()

,Time,V1,V2,V3,V4,V5,V6,V7,V8,V9,...,V21,V22,V23,V24,V25,V26,V27,V28,Amount,Class
0,0.0,-1.359807,-0.072781,2.536347,1.378155,-0.338321,0.462388,0.239599,0.098698,0.363787,...,-0.018307,0.277838,-0.110474,0.066928,0.128539,-0.189115,0.133558,-0.021053,149.62,0
1,0.0,1.191857,0.266151,0.166480,0.448154,0.060018,-0.082361,-0.078803,0.085102,-0.255425,...,-0.225775,-0.638672,0.101288,-0.339846,0.167170,0.125895,-0.008983,0.014724,2.69,0
2,1.0,-1.358354,-1.340163,1.773209,0.379780,-0.503198,1.800499,0.791461,0.247676,-1.514654,...,0.247998,0.771679,0.909412,-0.689281,-0.327642,-0.139097,-0.055353,-0.059752,378.66,0
3,1.0,-0.966272,-0.185226,1.792993,-0.863291,-0.010309,1.247203,0.237609,0.377436,-1.387024,...,-0.108300,0.005274,-0.190321,-1.175575,0.647376,-0.221929,0.062723,0.061458,123.50,0
4,2.0,-1.158233,0.877737,1.548718,0.403034,-0.407193,0.095921,0.592941,-0.270533,0.817739,...,-0.009431,0.798278,-0.137458,0.141267,-0.206010,0.502292,0.219422,0.215153,69.99,0


Interprétation :

L’aperçu des premières lignes confirme que le jeu de données a été correctement chargé. Le DataFrame contient 31 variables, parmi lesquelles `Time`, les variables anonymisées `V1` à `V28`, `Amount` et la variable cible `Class`.

Les premières observations affichées ont une valeur `Class` égale à `0`, ce qui correspond à des transactions non frauduleuses. Cette observation concerne uniquement les premières lignes et ne permet pas encore de conclure sur la répartition globale des transactions entre les deux classes.

Les variables `V1` à `V28` sont présentées sous forme de valeurs numériques, tandis que `Time` et `Amount` correspondent respectivement au temps de la transaction et à son montant.


 4. Aperçu des dernières lignes

Après avoir observé les premières observations du jeu de données, nous affichons également les dernières lignes afin de vérifier le contenu situé à la fin du dataset.

Cette vérification permet de compléter l’observation réalisée avec `df.head()` et de s’assurer que les dernières observations sont correctement présentes avant de poursuivre l’analyse de la structure du jeu de données.


In [10]:
df.tail()

,Time,V1,V2,V3,V4,V5,V6,V7,V8,V9,...,V21,V22,V23,V24,V25,V26,V27,V28,Amount,Class
284802,172786.0,-11.881118,10.071785,-9.834783,-2.066656,-5.364473,-2.606837,-4.918215,7.305334,1.914428,...,0.213454,0.111864,1.014480,-0.509348,1.436807,0.250034,0.943651,0.823731,0.77,0
284803,172787.0,-0.732789,-0.055080,2.035030,-0.738589,0.868229,1.058415,0.024330,0.294869,0.584800,...,0.214205,0.924384,0.012463,-1.016226,-0.606624,-0.395255,0.068472,-0.053527,24.79,0
284804,172788.0,1.919565,-0.301254,-3.249640,-0.557828,2.630515,3.031260,-0.296827,0.708417,0.432454,...,0.232045,0.578229,-0.037501,0.640134,0.265745,-0.087371,0.004455,-0.026561,67.88,0
284805,172788.0,-0.240440,0.530483,0.702510,0.689799,-0.377961,0.623708,-0.686180,0.679145,0.392087,...,0.265245,0.800049,-0.163298,0.123205,-0.569159,0.546668,0.108821,0.104533,10.00,0
284806,172792.0,-0.533413,-0.189733,0.703337,-0.506271,-0.012546,-0.649617,1.577006,-0.414650,0.486180,...,0.261057,0.643078,0.376777,0.008797,-0.473649,-0.818267,-0.002415,0.013649,217.00,0


Interprétation :

L’affichage des dernières lignes confirme que le jeu de données contient bien des observations jusqu’à l’indice `284806`. Les dernières observations présentent les mêmes variables que celles observées au début du jeu de données, soit 31 colonnes.

Dans les cinq dernières observations affichées, la variable `Class` prend la valeur `0`. Cela indique que ces observations correspondent à des transactions non frauduleuses. Cependant, cette observation porte uniquement sur les dernières lignes affichées et ne permet pas de déterminer la répartition globale des classes dans le jeu de données.
